config/externalname.go

```diff
@@ -252,11 +252,11 @@ var externalNameConfigs = map[string]config.ExternalName{
 	"google_compute_service_attachment": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/regions/{{ .parameters.region }}/serviceAttachments/{{ .external_name }}"),
 	// Note(donovanmuller): Requires organizational level permission 'compute.organizations.enableXpnHost'"
 	// Imported by using the following format: host-project-id
-	// "google_compute_shared_vpc_host_project": config.IdentifierFromProvider,
+	"google_compute_shared_vpc_host_project": config.IdentifierFromProvider,
 	// Note(donovanmuller): The google provider only supports this permission at project or organizational level currently
 	// It also requires access to multiple GCP Projects
 	// Imported by using the following format: host-project-id/service-project-id-1
-	// "google_compute_shared_vpc_service_project": TemplatedStringAsIdentifierWithNoName("{{ .setup.configuration.project }}/{{ .external_name }}"),
+	"google_compute_shared_vpc_service_project": TemplatedStringAsIdentifierWithNoName("{{ .setup.configuration.project }}/{{ .external_name }}"),
```

Comment:
I think we should be using the required `spec.forProvider.hostProject` (i.e., the `host_project` parameter) while constructing the ID instead of the project ID from the provider configuration (`.setup.configuration.project`). This is because in case these two differ (a different project ID is specified with `spec.forProvider.hostProject` than the project ID in the `ProviderConfig`) then the effective one should be the one under the `spec.forProvider`. Also `spec.forProvider.hostProject` is a required field. 
We could also consider making the `service_project` argument a naming argument but not sure if it's possible (or if it makes sense) to associate the same service project ID with different host project IDs. If this is not possible (or if this does not make sense), then something like the following should work:
```suggestion
	"google_compute_shared_vpc_service_project": config.TemplatedStringAsIdentifier("service_project", "{{ .parameters.host_project }}/{{ .external_name }}"),
```
This configuration will remove the `spec.forProvider.serviceProject` field and use `metadata.name` as the service project ID if an explicit external-name annotation is not set. And it becomes a little bit confusing if in the same cluster we would like to associate the same service project to multiple host projects (because you will then want to name multiple MRs with the same service project ID).

So, if this is a valid scenario (if one can associate the same service project with different host projects), we had better not treat `service_project` as a naming field. And in that case, I will suggest the `config.IdentifierFromProvider` instead:
```suggestion
	"google_compute_shared_vpc_service_project": config.IdentifierFromProvider,
```

If we are unsure, then `config.IdentifierFromProvider` is the safe choice here.